src/stan/mcmc/hmc/nuts/base_nuts.hpp

```diff
@@ -152,9 +151,9 @@ namespace stan {
         values.push_back(this->energy_);
       }
 
-      bool compute_criterion(Eigen::VectorXd& p_sharp_minus,
-                             Eigen::VectorXd& p_sharp_plus,
-                             Eigen::VectorXd& rho) {
+      virtual bool compute_criterion(Eigen::VectorXd& p_sharp_minus,
```

Comment:
Yes, for testing we need a derived class that overloads compute_criterion which is why I added it back in.